### PR TITLE
Fix Tests and Cleanup

### PR DESCRIPTION
**Pull request by avik-pal**
Fixes https://github.com/SciML/DiffEqFlux.jl/issues/781 Fixes https://github.com/SciML/DiffEqFlux.jl/issues/795 Fixes https://github.com/SciML/DiffEqFlux.jl/issues/639 Fixes https://github.com/SciML/DiffEqFlux.jl/issues/638, Fixes https://github.com/SciML/DiffEqFlux.jl/issues/707 Fixes https://github.com/SciML/DiffEqFlux.jl/issues/733

- [x] https://github.com/jonniedie/ComponentArrays.jl/pull/234
- [x] https://github.com/FluxML/Tracker.jl/pull/154
- [x] Aqua
- [x] CNF
- [x] Documentation Fix
- [ ] Maybe fix the DAEs

### CODE REVIEW

**Comment by prbzrg** [comment]
I think if we are removing `Flux`, we should also remove `Functors` and `trainable` dispatches.

**Comment by prbzrg** [comment]
I think https://github.com/JuliaDiff/ChainRules.jl/issues/713 is related.

**Comment by prbzrg** [comment]
Fixes #781

**Comment by prbzrg** [comment]
Fixing the AD problem closes https://github.com/SciML/DiffEqFlux.jl/issues/814

**Comment by prbzrg** [comment]
Supporting only Lux makes https://github.com/SciML/DiffEqFlux.jl/issues/795 irrelevant and closes it.

**Comment by prbzrg** [comment]
Now that we are breaking some APIs, should we consider https://github.com/SciML/DiffEqFlux.jl/issues/790 ?
IMO, having another function to generate `ODEProblem` and use it in the current function fixed issues like this that user wants control over internal information.

File: test/neural_ode_mm.jl [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
+    0 0 0]
+tspan = (0.0, 1.0)
+p = [0.04, 3e7, 1e4]
+func = ODEFunction(f; mass_matrix = M)
+prob = ODEProblem(func, u₀, tspan, p)
+sol = solve(prob, Rodas5(); saveat = 0.1)
+
+dudt2 = Chain(Dense(3 => 64, tanh), Dense(64 => 2))
+p, st = Lux.setup(rng, dudt2)
+p = ComponentArray(p)
```
Comment (ChrisRackauckas) [inline test/neural_ode_mm.jl:26]:
```suggestion
p = ComponentArray{Float64}(p)
```

**Comment by ChrisRackauckas** [comment]
> Maybe fix the DAEs

What's this part?

File: src/ffjord.jl [hunk 2]
```diff
@@ -36,214 +38,236 @@ References:
...
+
+function __ffjord(model, u, p, ad = AutoForwardDiff(), regularize::Bool = false,
+        monte_carlo::Bool = true)
+    N = ndims(u)
+    L = size(u, N - 1)
+    z = selectdim(u, N - 1, 1:(L - ifelse(regularize, 3, 1)))
+    if monte_carlo
+        mz, pb_f = Zygote.pullback(model, z, p)
+        e = CRC.@ignore_derivatives randn!(similar(mz))
+        eJ = first(pb_f(e))
```
[the remark this answers is not in the document]
Comment (avik-pal) [reply src/ffjord.jl:134]: I added `ADTypes` here so we can switch the internal AD using `ad = AutoForwardDiff`

**Comment by avik-pal** [comment]
> What's this part?

The DAEs didnot have any tests running even before, and I couldn't get AD to work currently

**Comment by ChrisRackauckas** [comment]
Which form? The mass matrix?

**Comment by avik-pal** [comment]
MM form (NeuralODEMM) works. the NeuralDAE form

File: src/ffjord.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (prbzrg) [reply src/ffjord.jl:134]: Using `JacVec` or `VecJac` from `SparseDiffTools`, we can pass `ad` to it.

[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply src/ffjord.jl:134]: Yeah what I mean is that it should be using JacVec (and almost never VecJac). It's e'Je, and so it should essentially always calculate e' * (Je) and not (e'J) * e which is what it does now (along with the PyTorch code)

**Pull request merged**